apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.cs

```diff
@@ -192,6 +191,14 @@ internal System.Type GetConfiguredClrType(SpannerConversionOptions options)
                 case TypeCode.Int64:
                     return typeof(long);
                 case TypeCode.Float64:
+                    if (options != null && options.UseSpannerNumericForDecimal)
```

Comment:
I don't think this is right, because we're not dealing with decimal - we're dealing with a double, inherently.

If someone wants a decimal -> numeric conversion by default, that doesn't mean they want float64 -> decimal.
I suspect we may need directional options, basically - which will be slightly more longwinded (but precise and predictable) for the customer to specify... it'll make this code simple though.